src/document_basic.c

```diff
@@ -220,14 +220,19 @@ int Document_LoadSchemaFieldJson(Document *doc, RedisSearchCtx *sctx, QueryError
   }
 
   RedisModule_CloseKey(k);
+  const char *keyName = RedisModule_StringPtrLen(doc->docKey, NULL);
 
-  RedisJSON jsonRoot = japi->openKeyWithFlags(ctx, doc->docKey, DOCUMENT_OPEN_KEY_INDEXING_FLAGS);
+  RedisJSON jsonRoot = NULL;
+  if (japi_ver >= 5) {
+    jsonRoot = japi->openKeyWithFlags(ctx, doc->docKey, DOCUMENT_OPEN_KEY_QUERY_FLAGS);
+  } else {
+    jsonRoot = japi->openKeyFromStr(ctx, keyName);
```

Comment:
Why is that the fallback? should we test that `japi_ver` supports `openKeyFromStr` ?